autoblocks/_impl/testing/evaluators/llm_judge.py

```diff
@@ -0,0 +1,186 @@
+import abc
+import json
+from textwrap import dedent
+from typing import Any
+from typing import Generic
+from typing import List
+from typing import Optional
+
+from autoblocks._impl import global_state
+from autoblocks._impl.config.constants import API_ENDPOINT
+from autoblocks._impl.testing.evaluators.util import get_autoblocks_api_key
+from autoblocks._impl.testing.evaluators.util import get_openai_client
+from autoblocks._impl.testing.evaluators.util import get_test_id
+from autoblocks._impl.testing.models import BaseTestEvaluator
+from autoblocks._impl.testing.models import Evaluation
+from autoblocks._impl.testing.models import EvaluatorOverride
+from autoblocks._impl.testing.models import EvaluatorOverrideComment
+from autoblocks._impl.testing.models import EvaluatorOverrideField
+from autoblocks._impl.testing.models import OutputType
+from autoblocks._impl.testing.models import ScoreChoice
+from autoblocks._impl.testing.models import TestCaseType
+from autoblocks._impl.testing.models import Threshold
+
+function_name = "select_answer"
+
+
+class BaseLLMJudge(BaseTestEvaluator, abc.ABC, Generic[TestCaseType, OutputType]):
+    """
+    Base evaluator for creating an LLM judge.
+    """
+
+    @property
+    def threshold(self) -> Optional[Threshold]:
+        """
+        The threshold for the evaluator.
+        """
+        return None
+
+    @property
+    def model(self) -> str:
+        """
+        The model to use for the evaluator.
+        It must be an OpenAI model that supports tools.
+        Defaults to "gpt-4-turbo".
+        """
+        return "gpt-4-turbo"
+
+    @property
+    def no_of_overrides(self) -> int:
+        """
+        The number of recent overrides to fetch for the evaluator and pass to make_prompt.
+
+        Defaults to 0.
+        """
+        return 0
+
+    @abc.abstractmethod
+    def make_prompt(
+        self, test_case: TestCaseType, output: OutputType, recent_overrides: List[EvaluatorOverride]
+    ) -> str:
+        """
+        The prompt passed to the LLM judge. Should be poised as a question.
+        """
+        pass
+
+    @property
+    @abc.abstractmethod
+    def score_choices(self) -> List[ScoreChoice]:
+        """
+        The choices for the LLM judge to use when answering.
+        """
+        pass
+
+    def _make_score_choice(self, value: float) -> Optional[ScoreChoice]:
+        return next((score for score in self.score_choices if score.value == value), None)
+
+    async def _get_recent_overrides(self) -> List[EvaluatorOverride]:
+        if self.no_of_overrides == 0:
+            # Don't fetch if the consumer hasn't requested any overrides
+            return []
+        test_id = get_test_id(evaluator_id=self.id)
+        resp = await global_state.http_client().get(
+            f"https://{API_ENDPOINT}/test-suites/{test_id}/evaluators/{self.id}/human-reviews?n={self.no_of_overrides}",
```

Comment:
you need to URL encode both the test ID and evaluator ID. there is a utility in `util.py`: https://github.com/autoblocksai/python-sdk/blob/dec5dcf332418e04e504e72a2d728b31c02fbd3c/autoblocks/_impl/util.py#L48